contracts/src/systems/end_game.cairo

```diff
@@ -0,0 +1,49 @@
+#[system]
+mod end_game_system {
+    use dojo::world::Context;
+    use starknet::ContractAddress;
+    use tsubasa::components::{Game, Outcome};
+    use tsubasa::events::{emit, EndGame};
+    use array::ArrayTrait;
+
+    fn execute(ctx: Context, game_id: felt252, scored_player: ContractAddress) {
+        let mut game = get!(ctx.world, game_id, Game);
+
+        // Reset round and update player score
+        game.turn = 0;
+        if (scored_player == game.player1) {
+            game.player1_score += 1;
+        } else if (scored_player == game.player2) {
+            game.player2_score += 1;
+        }
```

Comment:
this should be in the end turn logic ? After all the cards have attacked you'll know if someone won the round or not